phys2bids/bids.py

```diff
@@ -170,3 +172,64 @@ def use_heuristic(heur_file, sub, ses, filename, outdir, record_label=''):
     heurpath = os.path.join(fldr, f'{name}physio')
 
     return heurpath
+
+
+def participants_file(indir, outdir, yml, sub):
+    """[summary]
+
+    Args:
+        indir ([type]): [description]
+        outdir ([type]): [description]
+        yml ([type]): [description]
+        sub ([type]): [description]
+    """
+    LGR.info('Updating participants.tsv ...')
+    file_path = os.path.join(outdir, 'participants.tsv')
+    if not os.path.exists(file_path):
+        LGR.warning('phys2bids could not find participants.tsv')
+        # Read yaml info if file exists
+        if yml is not None and os.path.exists(os.path.join(indir, yml)):
+            LGR.info('Using yaml data to populate participants.tsv')
+            with open(os.path.join(indir, yml)) as f:
+                yaml_data = yaml.load(f, Loader=yaml.FullLoader)
+            p_id = sub
+            p_age = yaml_data['participant']['age']
+            p_sex = yaml_data['participant']['sex']
+            p_handedness = yaml_data['participant']['handedness']
+        else:
+            LGR.info('No yaml file was provided. Using phys2bids data to '
+                     'populate participants.tsv')
+            # Fill in with data from phys2bids
+            p_id = sub
+            p_age = 'n/a'
+            p_sex = 'n/a'
+            p_handedness = 'n/a'
+
+        # Write to participants.tsv file
+        header = ['participant_id', 'age', 'sex', 'handedness']
+        utils.append_list_as_row(file_path, header)
+
+        participants_data = [p_id, p_age, p_sex, p_handedness]
+        utils.append_list_as_row(file_path, participants_data)
+
+    else:  # If participants.tsv exists only update when subject is not there
+        LGR.info('phys2bids found participants.tsv. Updating if needed...')
+        # Find participant_id column in header
+        pf = open(file_path, 'r')
+        header = pf.readline().split("\t")
+        pf.close()
+        p_id_idx = header.index('participant_id')
+
+        # Check if subject is already in the file
+        sub_exists = False
+        with open(file_path) as pf:
+            tsvreader = reader(pf, delimiter="\t")
+            for line in tsvreader:
+                if sub in line[p_id_idx]:
+                    sub_exists = True
+                    break
+        # Only append to file if subject is not in the file
+        if not sub_exists:
+            LGR.info(f'Appending subjet {sub} to participants.tsv ...')
+            participants_data = [sub, 'n/a', 'n/a', 'n/a']
```

Comment:
What if the `participants.tsv` file didn't have four columns, but say, only 1 or 2?
I would add a check here for the number of 'n/a' to add.

Reply:
Good point!